client/src/ui/settings/SettingsMenu.hpp

```diff
@@ -125,5 +127,13 @@ namespace Engine
         std::vector<Graphics::Extent2u> _resolutions{{900, 600}, {1280, 720}, {1920, 1080}};
         std::size_t _currentResolution = 1; ///> Index of the current screen resolution
         bool _resolutionChanged = false;    ///> Flag indicating if the resolution has been changed
+
+        /**
+         * @brief List of available colorblind modes.
+         */
+        std::vector<Graphics::ColorBlindMode> _colorBlindModes{Graphics::ColorBlindMode::NONE,
```

Comment:
tu as déjà l'enum ColorBlindMode qui contient ces infos
tu peux faire juste une methode qui check et choisi le next mode, au lieu de créer ce vector:

```suggestion
static Graphics::ColorBlindMode nextMode(Graphics::ColorBlindMode m)
{
    switch (m) {
        case Graphics::ColorBlindMode::NONE: return Graphics::ColorBlindMode::DEUTERANOPIA;
        case Graphics::ColorBlindMode::DEUTERANOPIA: return Graphics::ColorBlindMode::PROTANOPIA;
        case Graphics::ColorBlindMode::PROTANOPIA: return Graphics::ColorBlindMode::TRITANOPIA;
        case Graphics::ColorBlindMode::TRITANOPIA: return Graphics::ColorBlindMode::NONE;
        default: return Graphics::ColorBlindMode::NONE;
    }
}```